api/v1alpha1/hostedcluster_types.go

```diff
@@ -36,6 +36,25 @@ type HostedClusterSpec struct {
 
 	// InfraID is used to identify the cluster in cloud platforms
 	InfraID string `json:"infraID,omitempty"`
+
+	// DNS configuration for the cluster
+	DNS DNSSpec `json:"dns,omitempty"`
+}
+
+// DNSSpec specifies the DNS configuration in the cluster
+type DNSSpec struct {
+	// BaseDomain is the base domain of the cluster.
```

Comment:
We need to figure out how to specify this unambiguously.

* What exactly does it represent? e.g.
  * The canonical domain suffix for any generated records for this cluster?
  * The suffix of a computed canonical suffix (e.g. name+basedomain)?
* Is it a canonical FQDN? If not, how is the canonical value computed?
  * The current implementation seems to imply this is not a canonical FQDN
* Must BaseDomain be a subdomain of the domain represented by PublicZoneId/PrivateZoneID?
* What are the uniqueness constraints?
* If there are uniqueness constraints, it's implied the value on spec is not usable absent some sort of async validation process
  * Where is that validation result surfaced on status?
  * What field will our controllers read given the spec value isn't necessarily valid?
  * Does this imply some sort of valid or admitted condition for HostedClusters generally the API should describe and the operator must honor?